proto/viam/app/v1/app.proto

```diff
@@ -1132,6 +1132,7 @@ message RegistryItem {
 
 message GetRegistryItemRequest {
   string item_id = 1;
+  optional bool include_markdown_documentation = 2;
```

Comment:
when do we not want to include markdown documentation?

Reply:
Actually for fleet we are never going to use this in the current scope. But it will be available if we decide to show markdown on the registry pages at some point. It also makes it available to external API callers if needed.I made it so that by default, markdown is not returned so that API payloads are smaller.

My plan is to return markdown in the internal builtins service (using the same technical approach as [Fancy Config Tech Spec](https://docs.google.com/document/d/1PMRix3pT0Wu7NE5A0NCCZ9YhQVaJY9p0-oKxKIAsMeI/edit?tab=t.0#heading=h.pkgijd3tsb7t)

The alternative of the bool is to make separate messages for Request messages and Response messages. I tried that in previous commits in this PR. Its really gross and has backwards compatibility issues.